bundles/org.openhab.binding.onecta/src/main/java/org/openhab/binding/onecta/internal/handler/OnectaBridgeHandler.java

```diff
@@ -0,0 +1,167 @@
+
+    private @Nullable OnectaConfiguration config;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private Units units = new Units();
+    private final OnectaConnectionClient onectaConnectionClient = new OnectaConnectionClient();
+
+    public Units getUnits() {
+        return units;
+    }
+
+    private @Nullable DeviceDiscoveryService deviceDiscoveryService;
+
+    /**
+     * Defines a runnable for a discovery
+     */
+    Runnable runnable = new Runnable() {
+        @Override
+        public void run() {
+            if (deviceDiscoveryService != null) {
+                deviceDiscoveryService.startScan();
+            }
+        }
+    };
+
+    public OnectaBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize.");
+        config = getConfigAs(OnectaConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        try {
+            onectaConnectionClient.startConnecton(thing.getConfiguration().get(CONFIG_PAR_USERID).toString(),
+                    thing.getConfiguration().get(CONFIG_PAR_PASSWORD).toString());
+
+            if (onectaConnectionClient.isOnline()) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+        } catch (DaikinCommunicationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, 10,
+                Integer.parseInt(thing.getConfiguration().get(CONFIG_PAR_REFRESHINTERVAL).toString()),
+                TimeUnit.SECONDS);
+
+        // Trigger discovery of Devices
+        scheduler.submit(runnable);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    private void pollDevices() {
+        logger.debug("pollDevices.");
+
+        if (getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+            try {
+                logger.debug("Try to restore connection ");
+                onectaConnectionClient.restoreConnecton();
+
+                if (onectaConnectionClient.isOnline()) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (DaikinCommunicationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Try to restore connection. See log for more information. ");
+            }
+        }
+
+        if (getThing().getStatus().equals(ThingStatus.ONLINE)) {
+
```

Comment:
```suggestion
```

I would not rely on the ThingStatus to determine if the connection has to be restored or not. The handler should not need to have deepr knowledge of the onectaConnectionClient.

The `onectaConnectionClient.refreshUnitsData();` method should have some sanity check and throw an exception like `DaikinCommunicationException` if there is an unrecoverable issue.